cmake/OpenCVCompilerOptimizations.cmake

```
@@ -375,11 +376,19 @@ elseif(PPC64LE)
   set(CPU_BASELINE "VSX" CACHE STRING "${HELP_CPU_BASELINE}")
 
 elseif(RISCV)
+  option(RISCV_RVV_SCALABLE "Use scalable RVV API on RISC-V" ON)
+
   ocv_update(CPU_RVV_TEST_FILE "${OpenCV_SOURCE_DIR}/cmake/checks/cpu_rvv.cpp")
   ocv_update(CPU_KNOWN_OPTIMIZATIONS "RVV")
-  ocv_update(CPU_RVV_FLAGS_ON "")
-  set(CPU_DISPATCH "RVV" CACHE STRING "${HELP_CPU_DISPATCH}")
-  set(CPU_BASELINE "RVV" CACHE STRING "${HELP_CPU_BASELINE}")
+  if(RISCV_RVV_SCALABLE)
+    ocv_update(CPU_RVV_FLAGS_ON "-march=rv64gcv -DCV_RVV_SCALABLE")
+  else()
+    ocv_update(CPU_RVV_FLAGS_ON "-march=rv64gcv")
+  endif()
```

```suggestion
  ocv_update(CPU_RVV_FLAGS_ON "-march=rv64gcv")
  if(RISCV_RVV_SCALABLE)
    set(CPU_RVV_FLAGS_ON "${CPU_RVV_FLAGS_ON} -DCV_RVV_SCALABLE" CACHE INTERNAL "")
  endif()
```
This way user won't forget to add `CV_RVV_SCALABLE` definition if they set custom `CPU_RVV_FLAGS_ON` line.

---

Got it.
I will check if we really need `CACHE INTERNAL` here tomorrow.